research/using-npm-link.md

```diff
@@ -0,0 +1,37 @@
+Получилось найти [одно решение](https://github.com/facebook/react/issues/13991#issuecomment-1867954003) где предлагалось в node\_modules прилинкованного пакета удалить React и линкануть его из node\_modules сервиса, в который подключили пакет (5 шаг в Solution).  
+Данное решение помогло решить проблему с этой странной ошибкой про хуки,  
+![Error text](./images/error-text.png)
+которая не решалась очень долго.  
+в данном [обсуждении](https://stackoverflow.com/questions/56663785/invalid-hook-call-hooks-can-only-be-called-inside-of-the-body-of-a-function-com)  мы впервые прочитали о связи этой ошибки с npm link и дублированием реакта
+
+Чтобы все заработало как надо в нашем случае, нужно провести несколько этапов танцев с бубнами:  
+terminal in react-tc-auth folder
+
+1. npm i  
+2. удалить папки node\_modules и es  
+3. nvm use 12 (без этого шаг 4 будет падать с проблемой выполнения npx simple-git-hooks)  
+4. npm ci
+
+terminal in React-Packages folder
+
+5. npm i (для установки общих пакетов для выполнения билда пакетов)
+
+terminal in react-tc-auth folder
+
+6. npm run build   
+7. nvm use 20.18 (повышаем ноду, чтобы link хранился в папке той же ноды, какая используется в проекте, куда подключаем пакет)  
+8. npm link   
+
+В сервисе, куда подключаем пакет:
```

Comment:
Add a link to this service where we tried it. Maybe even link these changes PR as an example.